src/box/lua/config/applier/box_cfg.lua

```diff
@@ -426,7 +441,82 @@ local function apply(config)
     box.cfg(box_cfg)
 end
 
+local function apply_names_schema_on_replace(old, new)
+    if old == nil and new[1] == 'replicaset_name' then
+        box.broadcast('apply_name', new[2])
+    end
+end
+
+local function apply_names_cluster_on_replace(old, new)
+    if old[3] == nil and new[3] ~= nil then
+        box.broadcast('apply_name', new[3])
+    end
+end
+
+local function apply_names_f(configdata)
+    local name = 'config_apply_names'
+    fiber.self():name(name)
+
+    box.ctl.wait_rw()
+
+    if table.equals(configdata._missing_names, {}) then
+        return
+    end
+
+    local names = configdata:names()
+    if configdata._missing_names[names.replicaset_name] then
+        box.space._schema:insert{'replicaset_name', names.replicaset_name}
```

Comment:
Technically the node may stop being RW right after the first yield. (in insert, for example).
So I suggest replacing all the names in a single transaction.

Reply:
Thank you for noticing! Fixed.